indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,419 @@
+                return (
+                    <Header key={content.label}>{content.label}</Header>);
+            case 'input':
+                return (
+                    <Field key={content.name}
+                           name={content.name}
+                           component={ReduxFormField}
+                           label={content.label}
+                           as="input" />);
+            case 'formgroup':
+                return (
+                    <Form.Group key={content.key}>
+                        {content.content.map(this.renderContent)}
+                    </Form.Group>);
+            case 'checkbox':
+                return (
+                    <Field key={content.name}
+                           name={content.name}
+                           component={ReduxCheckboxField}
+                           componentLabel={content.label}
+                           as={Checkbox} />);
+            case 'image':
+                return this.renderImage(room.spritePosition);
+            case 'attributes':
+                return this.renderAttributes();
+            case 'addButton':
+                return this.renderAddButton(content.editType);
+            case 'dropdown':
+                if (!attributes) {
+                    return;
+                }
+                options = this.generateOptions(attributes);
+                return (
+                    <Dropdown placeholder="Add new attributes"
+                              fluid
+                              multiple
+                              selection
+                              options={options}
+                              onChange={(event, values) => this.setState({newAttributes: values.value})} />
+                );
+        }
+    };
+
+
+    renderModalContent = (fprops) => {
+        const formProps = {onSubmit: fprops.handleSubmit};
+        return (
+            <>
+                <Modal.Header>
+                    <Translate>Edit Room Details</Translate>
+                </Modal.Header>
+                <Modal.Content>
+                    <Form id="room-form" {...formProps}>
+                        <Grid columns={3}>
+                            {columns.map(this.renderColumn)}
+                        </Grid>
+                    </Form>
+                </Modal.Content>
+                <Modal.Actions>
+                    <Button type="submit"
+                            form="room-form"
+                            primary>
+                        <Translate>Save</Translate>
+                    </Button>
+                </Modal.Actions>
+            </>
+        );
+    };
+
+    addAttributes = () => {
+        const {roomAttributes, attributes, newAttributes} = this.state;
+
+        const attributesObj = _.keyBy(attributes, 'name');
+        let newRoomAttributes = {};
+        newAttributes.map((attr) => {
+            newRoomAttributes = {...{[attributesObj[attr].name]: null}, ...newRoomAttributes};
```

Comment:
why reassign it all the time? If you want to build an object from an array, you can do it using one of these options:

- `_.fromPairs(newAttributes.map(attr => [attr.name, XXX]))`
- `newAttributes.reduce((obj, attr) => ({...obj, [attr.name]: XXX}), {});`